src/plugins/hooks.js

```diff
@@ -160,26 +165,35 @@ module.exports = function (Plugins) {
 		}
 		async.each(hookList, function (hookObj, next) {
 			if (typeof hookObj.method === 'function') {
-				var timedOut = false;
-
-				var timeoutId = setTimeout(function () {
+				let timedOut = false;
```

Comment:
Handling multiple callback calls requires only one more boolean like this.

Reply:
sure, and this is how what initially did, but I had to do that for all hook-types that handle callbacks, then i saw that i am writing redundant code, so I factored it out to `callable()` and I got carried away a little with the limiting and timeout options, but in defense, I had warning messages and all in there, which [you asked about too](https://github.com/NodeBB/NodeBB/pull/7496#issuecomment-499523315)

Anyways, I believe @julianlam reached a [consensus](https://github.com/NodeBB/NodeBB/pull/7496#issuecomment-499488440) that we will not be guarding against bad plugin code